crates/sui-types/src/committee.rs

```diff
@@ -61,14 +64,40 @@ impl Committee {
             // e.g. when a new validator is registering themself on-chain.
             .map(|(addr, _)| (*addr, (*addr).try_into().expect("Invalid Authority Key")))
             .collect();
+        let index_map: HashMap<_, _> = voting_rights
+            .iter()
+            .enumerate()
+            .map(|(index, (addr, _))| (*addr, index))
+            .collect();
+
         Ok(Committee {
             epoch,
             voting_rights,
             total_votes,
             expanded_keys,
+            index_map,
         })
     }
 
+    pub fn authorities_from_bitmap<'a>(
+        &'a self,
+        bitmap: &'a RoaringBitmap,
+    ) -> impl Iterator<Item = &AuthorityName> {
+        self.voting_rights
+            .iter()
+            .enumerate()
+            .filter(|&(i, _)| bitmap.contains(i as u32))
+            .map(|(_, (name, _))| name)
+    }
+
+    pub fn authority_index(&self, author: &AuthorityName) -> usize {
```

Comment:
We probably want to make this return `u32`, as this will break the map beyond the `u32` limit.

Reply:
Hashmap.get() returns usize, if this is the case, is this still a problem?

Reply:
`usize` is platform dependent, and may mean `u64`. 
This is probably not the first wrong thing about a 4GB-long table of authority indexes, but one thing it will not do is work with our `u32`-bound Bitmap.